### DISCUSSION TITLE
Show file content

### DISCUSSION THREAD

**Discussion by dya4i** [WinMerge/winmerge#1800] [General]
Hello, I have 2 folders with same files, but when I compare in winmerge, shows that are not same, I guess because name are different, how to solve this to compare by content, not name and make files alignment and show the are same, thanks.

**Comment by sdottaka** [comment]
Unfortunately, WinMerge does not have a built-in feature to automatically pair files with different names but identical contents during folder comparison. 
However, you can add a column that displays the hash values of the files (e.g., MD5) and group files with identical hash values but different names, as shown in the image below (in the example, the column is labeled "MD5(Move)").

![image](https://user-images.githubusercontent.com/98126/231506399-d255f958-2733-4d60-884a-8beb41cd3f63.png)

**Reply by emanueol** [reply]
> Unfortunately, WinMerge does not have a built-in feature to automatically pair files with different names but identical contents during folder comparison. However, you can add a column that displays the hash values of the files (e.g., MD5) and group files with identical hash values but different names, as shown in the image below (in the example, the column is labeled "MD5(Move)").
> 
> ![image](https://user-images.githubusercontent.com/98126/231506399-d255f958-2733-4d60-884a-8beb41cd3f63.png)

Hi @sdottaka as of 9-DEV-2024 i dont see the button "Additional Properties" :( I see those firest options like Filename, Fodler etc.. but dont find the MD5 ones on the list.
Have this option moved to somewhere else ? Im using version 2.16.8.0 X64 just installed few days ago.

Many Thanks.

**Reply by sdottaka** [reply]
2.16.8 was released four years ago.
The Additional Properties button has been added from 2.16.16, and the latest stable version is 2.16.44.

**Comment by dya4i** [comment]
Thank you for quick response, do you know any program with this options?
Try to help an old woman, I recovered lots of files, but it is hard to
check manually. Can I see files in 2 windows separately? And not in grup
like in my picture down?  And MD5 is empty, why?Thanks

[image: image.png]

<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

În mie., 12 apr. 2023 la 18:33, Takashi Sawanaka ***@***.***>
a scris:

**Comment by dya4i** [comment]
Now I have md5, after refresh, but show me only left one, not all files and
not in columns, I dont know how to compare and see duplicates and unique
ones.

<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

În mie., 12 apr. 2023 la 19:50, Ciorcalie Rodica ***@***.***> a
scris:

**Comment by sdottaka** [comment]
I cannot see the image attached in the email on GitHub, so I am not sure about your situation. However, if you are trying to find duplicate files, I believe it would be better to use a dedicated application like the one below, rather than WinMerge.

Duplicate Files Finder:  https://sourceforge.net/projects/doubles/
Czkawka: https://github.com/qarmin/czkawka/releases/tag/5.1.0
dupeGuru: https://dupeguru.voltaicideas.net/

[Duplicate Files Finder]
![image](https://user-images.githubusercontent.com/98126/231609978-e6bc8ef9-a269-474e-bae2-2e4756d0139d.png)
![image](https://user-images.githubusercontent.com/98126/231610010-05b0511d-aeaa-47f1-a173-259e24d46148.png)
